includes/admin/class-wp-job-manager-promoted-jobs-admin.php

```diff
@@ -114,6 +128,55 @@ public function add_action_notice( $actions_handled ) {
 		return $actions_handled;
 	}
 
+	/**
+	 * Add the host of the Promote Job form to the array of allowed redirect hosts.
+	 *
+	 * @param array $hosts Allowed redirect hosts.
+	 * @return array Updated array of allowed redirect hosts.
+	 */
+	public function add_to_allowed_redirect_hosts( $hosts ) {
+		$hosts[] = wp_parse_url( WP_Job_Manager_Helper_API::get_wpjmcom_url(), PHP_URL_HOST );
+		return $hosts;
+	}
+
+	/**
+	 * Handle the action to promote a job listing, validating as well as redirecting to the form on WPJobManager.com.
+	 *
+	 * @return void
+	 */
+	public function handle_promote_job() {
+		$post_id = absint( $_GET['post_id'] ?? 0 );
+		check_admin_referer( self::PROMOTE_JOB_ACTION . '-' . $post_id );
+		if ( ! $post_id ) {
+			wp_die( esc_html__( 'No job listing ID provided for promotion.', 'wp-job-manager' ), '', [ 'back_link' => true ] );
+		}
+		if ( ! current_user_can( 'manage_job_listings', $post_id ) || 'job_listing' !== get_post_type( $post_id ) ) {
+			wp_die( esc_html__( 'You do not have permission to promote this job listing.', 'wp-job-manager' ), '', [ 'back_link' => true ] );
+		}
+		$current_user = get_current_user_id();
+		$site_trust   = WP_Job_Manager_Site_Trust_Token::instance();
+		$token        = $site_trust->generate( 'user', $current_user );
+		if ( is_wp_error( $token ) ) {
+			wp_die( esc_html( $token->get_error_message() ) );
+		}
+		$site_url = home_url( '', 'https' );
+		$job_endpoint_url = rest_url( '/wpjm-internal/v1/promoted-jobs/' . $post_id, 'https' );
+		$job_endpoint_url = substr( $job_endpoint_url, strlen( $site_url ) );
+
+		$url = add_query_arg(
+			[
+				'user_id'  => $current_user,
+				'job_endpoint_url' => $job_endpoint_url,
+				'token'    => $token,
+				'site_url' => $site_url,
+				'locale'   => get_user_locale( $current_user ),
```

Comment:
I think there are some PHPCS issues here and on line 162.

Reply:
yep, it is on my TODO list to fix. Gonna fix it now. Thanks for the ping!

Reply:
Fixed in 73f1f2c and send PR #2507  to fix the `lint-staged` behavior so it catches this sort of mistake. Thanks!